modules/network/definition.nix

```
@@ -0,0 +1,60 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Module for Network Definitions
+#
+# The point of this module is to only store information about the network
+# configuration, and the logic that uses this information should be elsewhere.
+{lib, ...}: let
+  inherit (lib) mkOption types literalExpression;
+in {
+  options.ghaf.network.definition = let
+    networkSubmodule = types.submodule {
+      options = {
+        hostName = mkOption {
+          type = types.str;
+          description = ''
+            Hostname
+          '';
+        };
+        macAddress = mkOption {
+          type = types.str;
+          description = ''
+            MAC address
+          '';
+        };
+        ipAddressId = mkOption {
+          type = types.int;
+          description = ''
+            IP Address ID, the last octet of internal/debug network IP address
+          '';
+        };
+      };
+    };
+  in {
+    internalNetwork = mkOption {
+      description = "Internal network address";
```

Please clarify who's ip address this is:

i.e.

```suggestion
      description = "Internal network address of the host";
```

---

Ah. Hang on. This is not an ip address but a subnet!

```suggestion
      description = "Internal network subnet (/24)";
```

---

Sad to see no ipv6 here :)